### PR TITLE
swig bindings for Storage::Arch

### PR DESCRIPTION
**Pull request by wfeldt**
First try on getting bindings for Storage::Arch.

Not sure at all about the placement of that `#undef "_"`.

### CODE REVIEW

File: storage/Utils/AppUtil.h [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #include <map>
 #include <chrono>
 
+#undef _
```
Comment (aschnell) [inline storage/Utils/AppUtil.h]: Try to replace the #include "storage/Utils/XmlFile.h" by #include <libxml/tree.h> in Arch.h.

**Comment by aschnell** [comment]
You also have to update bindings/storage.deps.

File: storage/SystemInfo/Arch.cc [hunk 2]
```diff
@@ -71,19 +71,19 @@ namespace storage
 	if (is_ppc())
 	{
 	    AsciiFile cpuinfo("/proc/cpuinfo");
-	    vector<string>::const_iterator it = find_if(cpuinfo.lines(), string_starts_with("machine\t"));
+	    vector<std::string>::const_iterator it = find_if(cpuinfo.lines(), string_starts_with("machine\t"));
```
Comment (aschnell) [inline storage/SystemInfo/Arch.cc]: In .cc file you can use using std::string.

Comment (wfeldt) [inline storage/SystemInfo/Arch.cc]: the problem was that the generated storage_wrap.cxx does not know it

Comment (wfeldt) [inline storage/SystemInfo/Arch.cc]: sorry, you're probably right; the problem rather was that the make dependencies required a complete rebuild

File: storage/Utils/AppUtil.h (hunk 1 above)
Comment (wfeldt) [inline storage/Utils/AppUtil.h]: no luck

**Comment by aschnell** [comment]
Also please add Arch.h in doxygen.conf.

**Comment by aschnell** [comment]
Arch.h should also be included in the devel package now, so add pkginclude_HEADERS in storage/SystemInfo/Makefile.am.

**Comment by wfeldt** [comment]
replaced by https://github.com/openSUSE/libstorage-ng/pull/83

**Pull request closed without merging**